### ISSUE TITLE
Does not work with generic CSR dongle

### ISSUE THREAD

**Issue by gchehab** [thxomas/bdaddr#1]
Hi @thxomas, I have been trying to make it work with a generic CSR dongle. It reports the address has changed, appearing to have worked normally. However, after the device reset, the mac address remains the original one. Any clues or tips?

Tried with -r and even a temporary change -t, but all with the same results.

**Comment by thxomas** [comment]
Hi !
What is reported after the "Manufacturer :" word when you run it ?

**Comment by gchehab** [comment]
It reports:
```
$ sudo ./bdaddr -r -i hci0 <new mac>
Manufacturer: Cambridge Silicon Radio (10)
Device address: <old mac>
New BD address: <new mac>

Address changed - Device reset successfully

```
And the mac address is not changed. I am using the current main branch commit. I will try another dongle I have somewhere... 

Perhaps there is something odd on this specific unit or firmware version...

**Comment by thxomas** [comment]
Yes, looks like something specific. I'm sorry this part of the code was not written by me, and I have no CSR device to test.

**Comment by YusukeDS** [comment]
> Yes, looks like something specific. I'm sorry this part of the code was not written by me, and I have no CSR device to test.

Hello! Can you help me out?
With me im getting a "Can't write new address"..

**Comment by helgek** [comment]
I'm having the same issue:
Manufacturer:   Intel Corp. (2) 
Intel Corp. Bluetooth 9460/9560 Jefferson Peak (JfP)
(it's an Intel NUC device)

**Comment by helgek** [comment]
I searched for BLE capable USB dongles with the chipsets mentioned as supported but I couldn't find any. Do I assume correctly that only old bluetooth sticks are supported?

**Comment by KilianSSL** [comment]
I had the same issue like https://github.com/thxomas/bdaddr/issues/1#issuecomment-964477748 when i tried to reset the device using `bdaddr -r`, but when i use `hciconfig hci1 reset` it shows me the new bdaddr when running `hcitool dev`

**Comment by dreamlayers** [comment]
I guess it doesn't work with some CSR clones. Mine is an 0a12:0001 from AliExpress. The kernel warns me: `CSR: Unbranded CSR clone detected; adding workarounds and force-suspending once...`. Whatever I do I get `Can't write new address`.
